lib/matplotlib/pyplot.py

```diff
@@ -753,11 +754,11 @@ def figure(num=None,  # autoincrement if None, else integer from 1-N
     `~matplotlib.rcParams` defines the default values, which can be modified
     in the matplotlibrc file.
     """
-    if isinstance(num, Figure):
+    if isinstance(num, FigureBase):
         if num.canvas.manager is None:
             raise ValueError("The passed figure is not managed by pyplot")
         _pylab_helpers.Gcf.set_active(num.canvas.manager)
-        return num
+        return num.canvas.figure
```

Comment:
I'm not a fan of getting this info circularly from the canvas.  I think we should add `get_figure` to `FigureBase` and have it use `self._parent` to figure out the figure.

Reply:
Could just use `num.figure`?

https://github.com/matplotlib/matplotlib/blob/bb3519d4559149c8f29448ccbb9db9ea0474a4ef/lib/matplotlib/figure.py#L178

https://github.com/matplotlib/matplotlib/blob/bb3519d4559149c8f29448ccbb9db9ea0474a4ef/lib/matplotlib/figure.py#L2030

Reply:
Ah, yes, looks like that will work!

Reply:
I’ll try it tomorrow (I’m on iPad at the moment).

Reply:
Done.